diff --git a/node_modules/@changesets/cli/dist/cli.cjs.dev.js b/node_modules/@changesets/cli/dist/cli.cjs.dev.js
index f2a2bfc..e8eaf0e 100644
--- a/node_modules/@changesets/cli/dist/cli.cjs.dev.js
+++ b/node_modules/@changesets/cli/dist/cli.cjs.dev.js
@@ -1064,8 +1064,7 @@ async function run$1(input, flags, cwd) {
 
   if (!fs__default['default'].existsSync(path__default['default'].resolve(cwd, ".changeset"))) {
     logger.error("There is no .changeset folder. ");
-    logger.error("If this is the first time `changesets` have been used in this project, run `yarn changeset init` to get set up.");
-    logger.error("If you expected there to be changesets, you should check git history for when the folder was removed to ensure you do not lose any configuration.");
+    logger.error("Ensure you are running this command in the root of the repository.");
     throw new errors.ExitError(1);
   }
 
diff --git a/node_modules/@changesets/cli/dist/cli.cjs.prod.js b/node_modules/@changesets/cli/dist/cli.cjs.prod.js
index 5755f78..af498ea 100644
--- a/node_modules/@changesets/cli/dist/cli.cjs.prod.js
+++ b/node_modules/@changesets/cli/dist/cli.cjs.prod.js
@@ -320,7 +320,7 @@ async function version(cwd, options, config) {
 
 var isCI = !(!isCI__default.default && !process.env.GITHUB_ACTIONS);
 
-const npmRequestLimit = pLimit__default.default(40), npmPublishLimit = pLimit__default.default(10);
+const npmRequestLimit = pLimit__default.default(5), npmPublishLimit = pLimit__default.default(10);
 
 function jsonParse(input) {
   try {
@@ -592,8 +592,7 @@ async function pre(cwd, options) {
 async function run$1(input, flags, cwd) {
   if ("init" === input[0]) return void await init(cwd);
   if (!fs__default.default.existsSync(path__default.default.resolve(cwd, ".changeset"))) throw logger.error("There is no .changeset folder. "),
-  logger.error("If this is the first time `changesets` have been used in this project, run `yarn changeset init` to get set up."), 
-  logger.error("If you expected there to be changesets, you should check git history for when the folder was removed to ensure you do not lose any configuration."), 
+  logger.error("Ensure you are running this command in the root of the repository."),
   new errors.ExitError(1);
   const packages = await getPackages.getPackages(cwd);
   let config$1;
