diff --git a/node_modules/app-root-path/browser-shim.js b/node_modules/app-root-path/browser-shim.js
deleted file mode 100644
index 1730839..0000000
--- a/node_modules/app-root-path/browser-shim.js
+++ /dev/null
@@ -1,22 +0,0 @@
-'use strict';
-
-exports.path = require('path').dirname(require.main.filename);
-
-exports.resolve = function(pathToModule) {
-	return exports.path + pathToModule;
-};
-
-exports.require = function(pathToModule) {
-	var r = 'function' === typeof __webpack_require__
-		? __non_webpack_require__
-		: require;
-	return r(exports.resolve(pathToModule));
-};
-
-exports.toString = function() {
-	return exports.path;
-};
-
-exports.setPath = function(explicitlySetPath) {
-	exports.path = explicitlySetPath;
-};
\ No newline at end of file
